Week3/homework/index.js

```diff
@@ -0,0 +1,128 @@
+'use strict';
+
+const express = require('express');
+const mysql = require('mysql');
+const bodyParser = require('body-parser');
+
+const app = new express();
+
+app.use(bodyParser.json());
+
+const connection = mysql.createConnection({
+  host: 'localhost',
+  user: 'root',
+  password: 'password',
+  database: 'todo_database',
+});
+
+connection.connect();
+
+app.put(`/createList/:list`, (req, res) => {
+  const { list } = req.params;
+  connection.query(
+    `CREATE TABLE ${list} (todo_id INT(11) NOT NULL AUTO_INCREMENT,  todo varchar(50), done enum("true", "false"), user_id INT,  PRIMARY KEY (todo_id)  );`,
+    (error, results, fields) => {
+      if (error) {
+        console.log(error);
+        res.send(error);
+      } else {
+        res.send(results);
+      }
+    },
+  );
+});
+
+app.put(`/createUserList`, (req, res) => {
+  connection.query(
+    `CREATE TABLE User_List (user_id INT(11) NOT NULL AUTO_INCREMENT,  name varchar(50), email varchar(50),  PRIMARY KEY (user_id)  );`,
+    (error, results, fields) => {
+      if (error) {
+        console.log(error);
+        res.send(error);
+      } else {
+        res.send(results);
+      }
+    },
+  );
+});
+
+app.post(`/addUser`, (req, res) => {
+  const { name, email } = req.body;
+  connection.query(
+    `insert into User_List values (null, "${name}", "${email}"); `,
```

Comment:
Please use prepared statement, not direct string injection in the request otherwise your are vulnarable to hackers.

Reply:
okay I will.